### ISSUE TITLE
Getting Started Code

### ISSUE THREAD

**Issue by tik9** [Enhance-PHP/Enhance-PHP#18]
I run the Getting Started Code:
````
<?php

include_once('EnhanceTestFramework.php');
// Include your classes and test fixtures - they can be in separate files, just use include() statements for them!

class ExampleClass
{
    public function addTwoNumbers($a, $b)
    {
        return $a + $b;
    }
}
// Naming: By using "extends \Enhance\TestFixture" you signal that the public methods in // your class are tests.

class ExampleClassTests extends \Enhance\TestFixture
{
    private $target;

    public function setUp()
    {
        $this->target = \Enhance\Core::getCodeCoverageWrapper('ExampleClass');
    }
    public function addTwoNumbersWith3and2Expect5()
    {
        $result = $this->target->addTwoNumbers(3, 2);
        \Enhance\Assert::areIdentical(5, $result);
    }
}

Enhance\Core::runTests();

```

and get:

`Fatal error: Uncaught TypeError: get_parent_class(): Argument #1 ($object_or_class) must be an object or a valid class name, bool given in /Users/thome/php/public/EnhanceTestFramework.php on line 603`

Can you help to put me on the track?

**Comment by Steve-Fenton** [comment]
Does it work if you pop a namespace in there?

    namespace Example;

And then...

    $this->target = \Enhance\Core::getCodeCoverageWrapper('Example\ExampleClass');

**Comment by Steve-Fenton** [comment]
I think it may be because you haven't added the file to code spy, like this (insert your file name):

    \codespy\Analyzer::addFileToSpy('ExampleClass.php');
    Enhance\Core::runTests();

**Comment by tik9** [comment]
I get this back if including it: PHP Fatal error:  Uncaught Error: Class "codespy\Analyzer" not found in /Users/thome/php/public/enhance_test.php:31
I added it here:
\codespy\Analyzer::addFileToSpy('enhance_test.php');

My code with codespy and the namespace on first line is [here](https://github.com/tik9/php/blob/master/public/enhance_test.php)

**Comment by tik9** [comment]
I run it with php cli: `php public/enhance_test.php`

**Comment by Steve-Fenton** [comment]
Hi @tik9 

Can you try this for me, which I've tested in a branch of your repo...

Remove

    \codespy\Analyzer::addFileToSpy('enhance_test.php');

And adjust this line...

    Enhance\Core::runTests();

Needs to be fully qualified (start with \)

    \Enhance\Core::runTests();

I think there might be some update work to do for code coverage as it's not working in PHP 8 for me.

**Comment by Steve-Fenton** [comment]
Can you try one small adjustment...

When you call `public/enhance_test.php` can you "move to" the public folder and just run `enhance_test.php` to see if it makes a difference... otherwise, I'll take a new branch and have another look.

I wonder if Sandeep is still around to help us get the code spy working in the latest version of PHP? I'll tag @sras and see. I'm happy to dive in if needed, but he's expert in this area.

**Comment by tik9** [comment]
No, there is no difference when calling the script with php
enhance_test.php within public, I get same error:
Fatal error: Uncaught TypeError: get_parent_class(): ...

Am Di., 3. Mai 2022 um 15:46 Uhr schrieb Steve Fenton <
***@***.***>:

**Comment by tbreuss** [comment]
The above error should be fixed with #19.
